### DISCUSSION TITLE
Better system for zone sensitivity

### DISCUSSION THREAD

**Discussion by Aimtu321** [openbookings26/ContaCam#27] [Ideas]
Hi,

At the moment, in the "Recording" menu, you can select the sensitivity for motion detection. And if you want to reduce the sensitivity in certain areas, you can set, for example, 50% of the main sensitivity for an area in the "edit Zones" menu.

But isn't that unnecessarily complicated? Wouldn't it be more intuitive and practical to simply select the sensitivity value directly for the zones?

So, for example, remove the sensitivity selection in the "Recording" menu and just have an option for "Motion REC" or "Continuous REC". And in the "edit Zones" menu, you directly choose the sensitivity for the zones, and not a percentage of the main sensitivity, that you can then draw.

In my opinion, this is much better.

Furthermore, there's currently the problem of being somewhat limited. For example:

I have a zone that needs a sensitivity of 90. For another zone, a sensitivity of 70 or 80 would be ideal. But I can't set this because I can either set 100% main sensitivity or 50% main sensitivity, which would mean a sensitivity of either 90 or 45. If I could set the sensitivity directly for each zone, this problem wouldn't exist.

Please let me know what you think of this idea and if there's a realistic chance it could be implemented or if it can be programmed DIY.

![4](https://github.com/frakman1/ContaCam/assets/174721562/726bebc8-611e-429f-8676-d4cd71c77d31)

**Comment by openbookings26** [comment]
I agree. I always found the sensitivity settings to be bonkers and don't fully understand it myself. 
It's unlikely to change anytime soon but documenting how it currently works here will be helpful to anyone who wants to understand it better. Thank you.

**Comment by otpfiste** [comment]
Hi,

there is a reason why it's like this. In the beginning of ContaCam, all zones had the same sensitivity. So we only had a main sensitivity between 0..100. All was simple and intuitive. Most of the time people left it at 50 or lowered it to 40 if there were too many detections.

In some outdoor cases we had the problem of moving leaves. That was the main reason to introduce the relative sensitivity by zones. So we could exclude false detections by setting a lower relative sensitivity for the zones with moving leaves, but the main sensitivity behavior was the same and continued to work like usual for all the users which did not have a moving leave problem.

Why not make the zones sensitivity designer change directly the sensitivity? Because then we have to remove the main sensitivity, but 90% of the users do not know/care about setting zones sensitivity, so most users would be confused and forced to redraw all zones each time they just want to change the overall sensitivity. Moreover ContaCam has always been backwards compatible with older versions, such a change would break the compatibility with all previous versions. 

How the relative sensitivity works is [explained here](https://www.contaware.com/help/manual-faqs/17-03-contacam-settings/103-camera-advanced-settings.html) under the **Reduce false detections** section.

All the best,
Oliver

**Comment by Aimtu321** [comment]
Thanks for the answer!

Ok I understand. If 90% of the users don't set zones and only use the overall sensitivity then it makes no sense to remove it. But why not having both? You could let the overall sensitivity as it is but add another "recording mode" like "Motion REC with zones" or something like that (see image). If people want the overall sensitivity then they can choose it in the drop down like before but if you want to play with zones you can choose the mode "Motion REC with zones" and set the sensitivity in the zones menu.

I actually don't know what "backwards compatible" means in this case. And does it really have to be? What happens if ContaCam isn't compatible with older versions?

![1](https://github.com/user-attachments/assets/1caa7d9c-a84e-435b-b9cb-6a82e9caf90c)

Just an idea for the future maybe :)
